### PR TITLE
refactor: Migration for Tabbed Sql Editor

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY

This is the first portion of a larger bug issue of retaining a query's saved status when refreshing the page in sql lab. In order to create this we needed to create a foreign key in the tab state view that links it to the saved query. 

This migration does that, and the subsequent logic is here: https://github.com/apache/superset/pull/17771

Benchmark

```Current version of the DB is c53bae8f08dd

Running benchmark will downgrade the Superset DB to bb38f40aa3ff and upgrade to c53bae8f08dd again. There may be data loss in downgrades. Continue? [y/N]: y
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade c53bae8f08dd -> bb38f40aa3ff, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777

Identifying models used in the migration:
- ab_user (1 rows in table ab_user)
- dbs (1 rows in table dbs)
- query (0 rows in table query)
- tab_state (0 rows in table tab_state)
Benchmarking migration
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade bb38f40aa3ff -> c53bae8f08dd, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Migration on current DB took: 0.29 seconds
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade c53bae8f08dd -> bb38f40aa3ff, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Running with at least 10 entities of each model
- Adding 9 entities to the ab_user model
Processing ████████████████████████████████ 100%
- Adding 9 entities to the dbs model
Processing ████████████████████████████████ 100%
- Adding 10 entities to the query model
Processing ████████████████████████████████ 100%
- Adding 10 entities to the tab_state model
Processing ████████████████████████████████ 100%
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade bb38f40aa3ff -> c53bae8f08dd, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Migration for 10+ entities took: 0.29 seconds
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade c53bae8f08dd -> bb38f40aa3ff, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Running with at least 100 entities of each model
- Adding 90 entities to the ab_user model
Processing ████████████████████████████████ 100%
- Adding 90 entities to the dbs model
Processing ████████████████████████████████ 100%
- Adding 90 entities to the query model
Processing ████████████████████████████████ 100%
- Adding 90 entities to the tab_state model
Processing ████████████████████████████████ 100%
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade bb38f40aa3ff -> c53bae8f08dd, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Migration for 100+ entities took: 0.43 seconds
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade c53bae8f08dd -> bb38f40aa3ff, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Running with at least 1000 entities of each model
- Adding 900 entities to the ab_user model
Processing ████████████████████████████████ 100%
- Adding 900 entities to the dbs model
Processing ████████████████████████████████ 100%
- Adding 900 entities to the query model
Processing ████████████████████████████████ 100%
- Adding 900 entities to the tab_state model
Processing ████████████████████████████████ 100%
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade bb38f40aa3ff -> c53bae8f08dd, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Migration for 1000+ entities took: 0.28 seconds
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade c53bae8f08dd -> bb38f40aa3ff, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Running with at least 10000 entities of each model
- Adding 9000 entities to the ab_user model
Processing ████████████████████████████████ 100%
- Adding 9000 entities to the dbs model
Processing ████████████████████████████████ 100%
- Adding 9000 entities to the query model
Processing ████████████████████████████████ 100%
- Adding 9000 entities to the tab_state model
Processing ████████████████████████████████ 100%
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade bb38f40aa3ff -> c53bae8f08dd, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Migration for 10000+ entities took: 0.54 seconds
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade c53bae8f08dd -> bb38f40aa3ff, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Running with at least 100000 entities of each model
- Adding 90000 entities to the ab_user model
Processing ████████████████████████████████ 100%
- Adding 90000 entities to the dbs model
Processing ████████████████████████████████ 100%
- Adding 90000 entities to the query model
Processing ████████████████████████████████ 100%
- Adding 90000 entities to the tab_state model
Processing ████████████████████████████████ 100%
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade bb38f40aa3ff -> c53bae8f08dd, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Migration for 100000+ entities took: 0.51 seconds
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade c53bae8f08dd -> bb38f40aa3ff, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Running with at least 1000000 entities of each model
- Adding 900000 entities to the ab_user model
Processing ████████████████████████████████ 100%
- Adding 900000 entities to the dbs model
Processing ████████████████████████████████ 100%
- Adding 900000 entities to the query model
Processing ████████████████████████████████ 100%
- Adding 900000 entities to the tab_state model
Processing ████████████████████████████████ 100%
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade bb38f40aa3ff -> c53bae8f08dd, add_saved_query_foreign_key_to_tab_state
Revision ID: c53bae8f08dd
Revises: bb38f40aa3ff
Create Date: 2021-12-15 15:05:21.845777
Migration for 1000000+ entities took: 2.09 seconds
```

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by betodealmeida** [comment]
This looks great! But let's wait until we have the sibling PR ready and approved before merging this, in case we need to change the migration.

**Review by betodealmeida — CHANGES_REQUESTED** [review]
Blocking this just in case.

File: superset/migrations/versions/c53bae8f08dd_add_saved_query_foreign_key_to_tab_state.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# under the License.
+"""add_saved_query_foreign_key_to_tab_state
+Revision ID: c53bae8f08dd
+Revises: aea15018d53b
+Create Date: 2021-12-15 15:05:21.845777
+"""
+
+# revision identifiers, used by Alembic.
+revision = "c53bae8f08dd"
+down_revision = "bb38f40aa3ff"
```
Comment (eschutho) [inline superset/migrations/versions/c53bae8f08dd_add_saved_query_foreign_key_to_tab_state.py:25]: this down doesn't match line 19, fyi.

**Review by betodealmeida — APPROVED** [review]
Looks great!

**Pull request merged**